R/fda-table_02.R

```diff
@@ -247,3 +247,72 @@ make_table_02_tplyr <- function(df,
   }
   tbl
 }
+
+
+#' @describeIn make_table_02 Create FDA table 2 using functions from `gtsummary`.
+#'
+#' @return
+#' * `make_table_02_gt` returns a `tbl_summary` object
+#'
+#' @examples
+#'
+#' advs <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "advs")
+#' adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl") %>%
+#'   mutate(AGEGR1 = as.factor(case_when(
+#'     AGE >= 17 & AGE < 65 ~ ">=17 to <65",
+#'     AGE >= 65 ~ ">=65",
+#'     AGE >= 65 & AGE < 75 ~ ">=65 to <75",
+#'     AGE >= 75 ~ ">=75"
+#'   )) %>%
+#'     formatters::with_label("Age Group, years"))
+#'
+#' advs <- advs %>%
+#'   dplyr::filter(AVISIT == "BASELINE", VSTESTCD == "TEMP") %>%
+#'   dplyr::select("USUBJID", "AVAL")
+#'
+#' anl <- dplyr::left_join(adsl, advs, by = "USUBJID") %>%
+#'   formatters::var_relabel(
+#'     AGE = "Age, years",
+#'     AVAL = "Baseline Temperature (C)"
+#'   )
+#'
+#' tbl <- make_table_02_gt(df = anl)
+#' tbl
+#'
+#' @export
+make_table_02_gt <- function(df,
+                             alt_counts_df = NULL,
+                             show_colcounts = TRUE,
+                             arm_var = "ARM",
+                             vars = c("SEX", "AGE", "AGEGR1", "RACE", "ETHNIC", "COUNTRY"),
+                             lbl_vars = formatters::var_labels(df, fill = TRUE)[vars],
+                             lbl_overall = "Total Population",
+                             annotations = NULL) {
+  checkmate::assert_subset(c("SAFFL", vars, arm_var), names(df))
+  assert_flag_variables(df, "SAFFL")
+
+  df <- df %>%
+    filter(SAFFL == "Y") %>%
+    select(all_of(c(vars, arm_var)))
+
+  alt_counts_df <- alt_counts_df_preproc(alt_counts_df, arm_var)
+
+  tbl <- df %>%
+    tbl_summary(
+      by = arm_var,
+      type = all_continuous() ~ "continuous2",
+      statistic = list(
+        all_continuous() ~ c(
+          "{mean} ({sd})",
+          "{median} ({min} - {max})"
+        ),
+        all_categorical() ~ "{n} ({p}%)"
+      ),
+      digits = all_continuous() ~ 2,
+    ) %>%
+    add_overall(last = TRUE, col_label = lbl_overall) %>%
+    modify_header(all_stat_cols() ~ "**{level}**  \n N = {n}") %>%
```

Comment:
Swap these lines so that `lbl_overall` is used
```suggestion
    modify_header(all_stat_cols() ~ "**{level}**  \n (N={n})") %>%
    add_overall(last = TRUE, col_label = paste0("**", lbl_overall, "**  \n (N={n})")) %>%
```